src/libraries/System.Runtime/tests/System.IO.FileSystem.Tests/File/Copy.cs

```diff
@@ -422,6 +422,121 @@ public void DestinationFileIsTruncatedWhenItsLargerThanSourceFile()
+        [InlineData(".leading", "destination")]
+        [InlineData("source", ".leading")]
+        public void CopyWithLeadingDots(string sourceFileName, string destFileName)
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string sourcePath = Path.Combine(testDir.FullName, sourceFileName);
+            string destPath = Path.Combine(testDir.FullName, destFileName);
+
+            File.Create(sourcePath).Dispose();
+            Copy(sourcePath, destPath);
+
+            Assert.True(File.Exists(sourcePath));
+            Assert.True(File.Exists(destPath));
+        }
+
+        [Theory]
+        [InlineData("-", "destination")]
+        [InlineData("source", "-")]
+        [InlineData("--", "destination")]
+        public void CopyWithDashPrefixedNames(string sourceFileName, string destFileName)
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string sourcePath = Path.Combine(testDir.FullName, sourceFileName);
+            string destPath = Path.Combine(testDir.FullName, destFileName);
+
+            File.Create(sourcePath).Dispose();
+            Copy(sourcePath, destPath);
+
+            Assert.True(File.Exists(sourcePath));
+            Assert.True(File.Exists(destPath));
+        }
+
+        [Theory]
+        [InlineData("file\tname", "destination")]
+        [InlineData("source", "file\tname")]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public void UnixCopyWithEmbeddedControlCharacters(string sourceFileName, string destFileName)
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string sourcePath = Path.Combine(testDir.FullName, sourceFileName);
+            string destPath = Path.Combine(testDir.FullName, destFileName);
+
+            File.Create(sourcePath).Dispose();
+            Copy(sourcePath, destPath);
+
+            Assert.True(File.Exists(sourcePath));
+            Assert.True(File.Exists(destPath));
+        }
+
+        [ConditionalTheory(nameof(UsingNewNormalization))]
+        [InlineData("trailing ", "destination")]
+        [InlineData("source", "trailing ")]
+        [InlineData("trailing.", "destination")]
+        [InlineData("source", "trailing.")]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void WindowsCopyWithTrailingSpacePeriod_ViaExtendedSyntax(string sourceFileName, string destFileName)
+        {
+            // On Windows, files with trailing spaces/periods must use \\?\ prefix for creation
+            // because path normalization strips these characters. While enumeration APIs like
+            // Directory.GetFiles can discover these files, direct string-based APIs (File.OpenRead,
+            // File.Exists) will fail because the trailing characters get stripped during normalization.
+            // Only FileInfo/DirectoryInfo objects from enumeration use EnsureExtendedPrefixIfNeeded
```

Comment:
`EnsureExtendedPrefixIfNeeded` is used by nearly all System.IO file APIs. It is not what makes `FileInfo/DirectoryInfo` work fine.

`FileInfo/DirectoryInfo` work fine since their implementation skips the path normalization.

Please update the comment